### PR TITLE
Use azure-identity to log into Azure.

### PR DESCRIPTION
**Pull request by dastrupl**

### CODE REVIEW

File: prompterator/models/openai_models.py [hunk 1]
```diff
@@ -57,6 +58,25 @@ def __init__(self):
             self.client = AzureOpenAI(
                 api_version=api_version, azure_endpoint=endpoint, api_key=api_key
             )
+        elif self.openai_variant == "azure-default-credentials":
+            default_credential = DefaultAzureCredential()
+            token = default_credential.get_token(
+                "https://cognitiveservices.azure.com/.default")
```
Comment (mrshu) [inline prompterator/models/openai_models.py]: @dastrupl could we please extract this into an environment variable and add a bit of documentation to `README.md` about it? As you can easily imagine, people might ask for different scopes depending on their Azure tenants 🙂

**Review by mrshu — CHANGES_REQUESTED** [review]
Thanks a bunch for the PR @dastrupl! There is only one small thing that I think would make sense to update. Would you mind taking a look at it?

Thanks!

File: prompterator/models/openai_models.py (hunk 1 above)
Comment (dastrupl) [reply prompterator/models/openai_models.py]: Do you mean to get rid of the special treatment of "azure-default-credentials" via separate model types and control only the credentials via an env variable?

Comment (dastrupl) [reply prompterator/models/openai_models.py]: 2 new commits added to address. Please let me know if it is Ok now.

File: README.md [hunk 2]
```diff
@@ -47,6 +47,8 @@ If you use PyCharm, consider storing these in your
   OpenAI APIs.
 - `AZURE_OPENAI_API_KEY` and `AZURE_OPENAI_API_BASE`: Optional. Only if you want to use OpenAI
   models (ChatGPT, GPT-4, etc.) via Azure OpenAI APIs.
+  - `AZURE_OPENAI_USE_DEFAULT_CREDENTIALS`: Optional. If set to `True`, the Azure OpenAI API will use
+    the default credentials from the environment, as per the (https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python).
```
Comment (mrshu) [inline README.md]:
```suggestion
    the default credentials from the environment, as per https://learn.microsoft.com/en-us/python/api/overview/azure/identity-readme?view=azure-python
```

**Review by mrshu — APPROVED** [review]
Thanks for this @dastrupl -- it looks good!

File: README.md (hunk 2 above)
Comment (mrshu) [reply README.md]: Only a small suggestion in here :)

**Pull request merged**